### ISSUE TITLE
azurerm_linux_function_app outbound_ip_address attributes are empty

### ISSUE THREAD

**Issue by psethwick** [hashicorp/terraform-provider-azurerm#14489]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request

### Terraform (and AzureRM Provider) Version

* Terraform Core version: v1.0.11
* AzureRM Provider version: v2.87.0

### Terraform Configuration Files

```hcl
# Copy-paste your Terraform configurations here - for large Terraform configs,
# please use a service like Dropbox and share a link to the ZIP file. For
# security, you can also encrypt the files using our GPG public key: https://keybase.io/hashicorp
```

### Description / Feedback

I have an azurerm_linux_function_app deployed.

Here are the ip address attributes from `state show`:
```
    outbound_ip_address_list          = []
    outbound_ip_addresses             = ""
    possible_outbound_ip_address_list = []
    possible_outbound_ip_addresses    = ""
```
I think these should have the values I can see in the azure portal.

**Comment by sathishmohan-ps** [comment]
This is happening for **azurerm_windows_function_app** as well. 

I am using following versions.

Terraform Core version: v1.1.6
AzureRM Provider version: v3.0.2

Can this be fixed sooner ?

**Comment by erorubrz** [comment]
Same for me. Using **azurerm_windows_function_app** as well. I had to downgrade to 2.x again, since we use this to automate FW rules creation.

**Comment by erorubrz** [comment]
Same for me. Using **azurerm_windows_function_app** and azurerm 3.x. No issues were observed on 2.x which I needed to downgrade again.

Terraform Core version: v1.1.9
AzureRM Provider version: v3.3.0

**Comment by johanandrel** [comment]
Same issue here using a **azurerm_linux_function_app**:

Terraform Core version: v1.1.4
AzureRM Provider version: v3.7.0

**Comment by dptaf** [comment]
seeing this as well on linux_function_app. any thoughts?

**Comment by ericdunn** [comment]
> seeing this as well on linux_function_app. any thoughts?

I ended up reverting back to 2.9.5. We use these in another resource for firewall restrictions, so we really can't afford to not have them in place. Manually entering all of those IPs is also a pain and something that would get wiped away each time.  When it works, we'll upgrade.

**Comment by evensky** [comment]
The linux function app still has empty outbound ip address lists:

Terraform Core version: v1.2.6
AzureRM Provider version: v3.10.0

We're committed to 3.x features of the provider, so it looks like we'll have to manually add these. Ugh.

**Comment by arnehilmann** [comment]
Still a problem, unfortunately...

Terraform v1.2.7
+ provider registry.terraform.io/hashicorp/azurerm v3.18.0

**Comment by Davidh-claranet** [comment]
Fixed in Azurerm 3.19
https://github.com/hashicorp/terraform-provider-azurerm/pull/17361

**Comment by tombuildsstuff** [comment]
Fixed via #17361 - thanks for the link @Davidh-claranet
